### PR TITLE
update kaminari dependency to 1.x

### PR DESCRIPTION
**Pull request by jrochkind**
https://github.com/kaminari/kaminari/blob/master/CHANGELOG.md

### CODE REVIEW

File: core/solidus_core.gemspec [hunk 1]
```diff
@@ -28,7 +28,7 @@ Gem::Specification.new do |s|
   s.add_dependency 'ffaker', '~> 2.0'
   s.add_dependency 'friendly_id', '~> 5.0'
   s.add_dependency 'highline', '~> 1.7' # Necessary for the install generator
-  s.add_dependency 'kaminari', '~> 0.15', '>= 0.15.1'
+  s.add_dependency 'kaminari', '~> 1.0'
```
Comment (tvdeyen) [inline core/solidus_core.gemspec]: We are forcing stores to use Kaminari 1.0 then (which has some breaking changes), while `['>= 0.17', '< 2.0']` would be less intrusive and still Rails 5 compatible.

Comment (jrochkind) [inline core/solidus_core.gemspec]: Sure, that makes sense, I'll change it to be that.  (I'm not sure it actually has any intentional breaking changes, I couldn't find any; but may have unintentional breaking changes and bugs; and I agree with the principle to not force people to upgrade)

This _was_ allowing as low as `0.15.1` before, does that not work on Rails5 at all, I should require at least

Comment (tvdeyen) [inline core/solidus_core.gemspec]: https://github.com/kaminari/kaminari/blob/master/CHANGELOG.md#breaking-changes

The most concerning one is the removal of `num_pages` on AR scopes.

Thanks

Comment (jrochkind) [inline core/solidus_core.gemspec]: done.

**Pull request merged**